task/task_definition_parameters.go

```diff
@@ -0,0 +1,227 @@
+	Operator string `yaml:"operator"`
+}
+
+// This struct contains the rule.
+type Rule struct {
+	// Metic indicates the name of the metric. These can be:
+	// 	Cpu
+	//	Mem
+	//	Shard
+	Metric string `yaml:"metric"`
+	// Limit indicates the threshold value for a metric.
+	// If this threshold is achieved for a given metric for the decision periond then the rule will be activated.
+	Limit float32 `yaml:"limit"`
+	// Stat indicates the statistics on which the evaluation of the rule will happen.
+	// For Cpu and Mem the values can be:
+	//		Avg: The average CPU or MEM value will be calculated for a given decision period.
+	//		Count: The number of occurences where CPU or MEM value crossed the threshold limit.
+	//		Term:
+	// For rule: Shard, the stat will not be applicable as the shard will be calculated across the cluster and is not a statistical value.
+	Stat string `yaml:"stat"`
+	// DecisionPeriod indicates the time in minutes for which a rule is evalated.
+	DecisionPeriod int `yaml:"decision_period"`
+	// Occurences indicate the number of time a rule reached the threshold limit for a give decision period.
+	// It will be applicable only when the Stat is set to Count.
+	Occurences int `yaml:"occurences"`
+}
+
+// This struct contains the task details which is set of actions.
+type TaskDetails struct {
+	// Tasks indicates list of task.
+	// A task indicates what operation needs to be recommended by recommendation engine.
+	// As of now tasks can be of two types:
+	//
+	//	scale_up_by_1
+	//	scale_down_by_1
+	Tasks []Task `yaml:"action"`
+}
+
+// Inputs:
+// Caller: Object of TaskDetails
+// Description:
+//
+//		EvaluateTask will go through all the tasks one by one. and
+//		It check if the task are meeting the criteria based on rules and operator.
+//		If the task is meeting the criteria then it will push the task to recommendation queue.
+//
+// Return:
+
+func (t TaskDetails) EvaluateTask() {
+	for _, v := range t.Tasks {
+		isRecommendeTask := v.GetNextTask()
+		if isRecommendeTask {
+			v.PushToRecommendationQueue()
+		} else {
+			fmt.Printf("The %s task is not recommended as rules are not satisfied", v.TaskName)
+			fmt.Println()
+		}
+	}
+}
+
+// Inputs:
+// Caller: Object of Task
+// Description:
+//
+//		GetNextTask will get the Task and iterate through all the rules inside the task.
+//		Based on the operator it will check if it should iterate through all the rules or not.
+//		It will call GetNextRule while iterating through the rules.
+//		Based on the result GetNextTask will check if a task can be recommended or not.
+//
+// Return:
+//
+//		Return if a task can be recommended or not(bool)
+
+func (t Task) GetNextTask() bool {
+	var isRecommendedTask bool = true
+	var isRecommendedRule bool
+
+	if t.Operator == "AND" {
+		// Here we will add go routine.
+		// So that all the rules getMetrics will be fetched in concurrent way
+		for _, v := range t.Rules {
+			isRecommendedRule = v.GetNextRule()
```

Comment:
We should have a mechanism to check if we have enough data points for evaluating the rules. In case of AND if we do not have enough data point for even one rule then the for loop should be broken. This can be considered as ToDo.